src/parser/ui/DonutChart.tsx

```diff
@@ -69,56 +95,24 @@ class DonutChart extends PureComponent<Props> {
         )}
       </div>
     );
-  }
-  renderChart(items: Item[], chartSize: number, innerRadiusFactor: number) {
-    const innerRadius = chartSize * innerRadiusFactor;
+  };
+
+  const renderChart = () => {
+    const data = { items };
```

Comment:
this should also be memoized. i know it wasn't before either but it should've been and we should fix it while we're touching this code.